open-budget/app/scripts/models.coffee

```diff
@@ -453,15 +453,77 @@ class Entity extends Backbone.Model
                 supports: []
                 exemptions: []
                 id: null
+                exemptions_by_publisher: {}
+                exemptions_sum: null
 
         initialize: (options) ->
                 @pageModel = options.pageModel
+                @entity_id = options.entityId
 
         doFetch: ->
-                @fetch(dataType: @pageModel.get('dataType'))
-
-        url: ->
-                "#{pageModel.get('baseURL')}/api/entity/#{pageModel.get('entityId')}"
+                @fetch(dataType: @pageModel.get('dataType'), success: @handleFetchResult)
+
+        url: =>
+                "#{pageModel.get('baseURL')}/api/entity/#{@entity_id}"
+
+        handleFetchResult: (collection, response) =>
+            @supports = response.supports
+            @exemptions = response.exemptions
+
+            @set('exemptions_sum', @get_exemptions_total_volume())
+
+            @exemptionsByPublisher()
+
+            @trigger('ready')
+
+        get_exemptions_total_volume: =>
+            exemptions_sum = 0
+            for exemption in @exemptions
+                exemptions_sum += exemption.volume
+            return exemptions_sum
+
+        exemptionsByPublisher: =>
+            exemptions_by_publisher = {}
+            for exemption in @exemptions
+                if not exemptions_by_publisher[exemption.publisher]?
+                    exemptions_by_publisher[exemption.publisher] = {publisher: exemption.publisher, exemptions: [], total_volume: 0}
+                exemptions_by_publisher[exemption.publisher].exemptions.push(exemption)
+                exemptions_by_publisher[exemption.publisher].total_volume += exemption.volume
+                exemptions_by_publisher[exemption.publisher].start_date = @min_date(exemptions_by_publisher[exemption.publisher].start_date, @convert_str_to_date(exemption.start_date))
+                exemptions_by_publisher[exemption.publisher].end_date = @max_date(exemptions_by_publisher[exemption.publisher].end_date, @convert_str_to_date(exemption.end_date))
+
+            for publisher of exemptions_by_publisher
+                if (exemptions_by_publisher.hasOwnProperty(publisher))
+                    exemptions_by_publisher[publisher].start_date = @convert_date_to_str(exemptions_by_publisher[publisher].start_date)
+                    exemptions_by_publisher[publisher].end_date = @convert_date_to_str(exemptions_by_publisher[publisher].end_date)
+
+            exemptions_by_publisher
+
+
+        convert_str_to_date: (date_str) ->
```

Comment:
let's move these utility functions to the utility method container (which is named 'hello.coffee' for historical reasons :smile: )